aten/src/ATen/cuda/tunable/Tunable.cpp

```diff
@@ -107,6 +107,34 @@ void TuningResultsManager::Add(const std::string& op_signature, const std::strin
   AddImpl(op_signature, params_signature, best, it->second);
 }
 
+void TuningResultsManager::RecordUntuned( std::ofstream& untuned_file, const std::string& op_signature, const std::string& params_signature) {
+  std::scoped_lock l{lock_};
+  if (!untuned_file.good()){
+    TORCH_WARN("failed to open file for writing; untuned gemm will not be saved");
+  }
+  else{
+    bool isNew = false;
+    auto it = untuned_results_.find(op_signature);
+    if (it == untuned_results_.end()) {
+      it = untuned_results_.insert({op_signature, {}}).first;
+      isNew = true;
+    }
+
+    auto it_kernel_map = it->second.find(params_signature);
+    if (it_kernel_map == it->second.end())
+    {
+      it->second.insert(params_signature);
+      isNew = true;
+    }
+
+    if (isNew){
+      std::string device = c10::str(int(c10::cuda::current_device()));
+      untuned_file << "Untuned GPU " <<device<<","<< op_signature << "," << params_signature << std::endl;
```

Comment:
```suggestion
      untuned_file << op_signature << "," << params_signature << std::endl;
```
The filename you construct will already have the device ordinal in the filename.  No need to repeat it for each line of the file.  You also don't need "Untuned GPU " text on each line of the file -- it's implied that your untuned file will be full of "untuned" entries; this is redundant.